go/vt/vtgate/bsonp3vtgateservice/server.go

```diff
@@ -197,212 +186,141 @@ func (vtg *VTGateP3) ExecuteBatchKeyspaceIds(ctx context.Context, request *pb.Ex
 	ctx = callerid.NewContext(ctx,
 		request.CallerId,
-		callerid.NewImmediateCallerID("gorpc client"))
+		callerid.NewImmediateCallerID("bsonp3 client"))
 	vtgErr := vtg.server.StreamExecuteKeyRanges(ctx,
 		string(request.Query.Sql),
 		tproto.Proto3ToBindVariables(request.Query.BindVariables),
 		request.Keyspace,
 		request.KeyRanges,
 		request.TabletType,
-		func(reply *proto.QueryResult) error {
+		func(value *proto.QueryResult) error {
 			return sendReply(&pb.StreamExecuteKeyRangesResponse{
-				Error:  vtgate.VtGateErrorToVtRPCError(nil, reply.Error),
-				Result: mproto.QueryResultToProto3(reply.Result),
+				Result: mproto.QueryResultToProto3(value.Result),
 			})
 		})
-	if vtgErr == nil {
-		return nil
-	}
-	if *vtgate.RPCErrorOnlyInReply {
-		// If there was an app error, send a QueryResult back with it.
-		// Sending back errors this way is not backwards compatible. If a (new) server sends an additional
-		// QueryResult with an error, and the (old) client doesn't know how to read it, it will cause
-		// problems where the client will get out of sync with the number of QueryResults sent.
-		// That's why this the error is only sent this way when the --rpc_errors_only_in_reply flag is set
-		// (signalling that all clients are able to handle new-style errors).
-		return sendReply(&pb.StreamExecuteKeyRangesResponse{
-			Error: vtgate.VtGateErrorToVtRPCError(vtgErr, ""),
-		})
-	}
-	return vtgErr
+	return vterrors.ToJSONError(vtgErr)
 }
 
-// Begin2 is the RPC version of vtgateservice.VTGateService method
-func (vtg *VTGateP3) Begin2(ctx context.Context, request *pb.BeginRequest, response *pb.BeginResponse) (err error) {
+// Begin is the RPC version of vtgateservice.VTGateService method
+func (vtg *VTGateP3) Begin(ctx context.Context, request *pb.BeginRequest, response *pb.BeginResponse) (err error) {
 	defer vtg.server.HandlePanic(&err)
 	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(*rpcTimeout))
 	defer cancel()
 	ctx = callerid.NewContext(ctx,
 		request.CallerId,
-		callerid.NewImmediateCallerID("gorpc client"))
-	// Don't pass in a nil pointer
-	session := &proto.Session{}
-	vtgErr := vtg.server.Begin(ctx, session)
-	response.Session = proto.SessionToProto(session)
-	response.Error = vtgate.VtGateErrorToVtRPCError(vtgErr, "")
-	if *vtgate.RPCErrorOnlyInReply {
+		callerid.NewImmediateCallerID("bsonp3 client"))
+	outSession := new(proto.Session)
+	vtgErr := vtg.server.Begin(ctx, outSession)
+	if vtgErr == nil {
+		response.Session = proto.SessionToProto(outSession)
 		return nil
 	}
-	return vtgErr
+	return vterrors.ToJSONError(vtgErr)
 }
 
-// Commit2 is the RPC version of vtgateservice.VTGateService method
-func (vtg *VTGateP3) Commit2(ctx context.Context, request *pb.CommitRequest, response *pb.CommitResponse) (err error) {
+// Commit is the RPC version of vtgateservice.VTGateService method
+func (vtg *VTGateP3) Commit(ctx context.Context, request *pb.CommitRequest, response *pb.CommitResponse) (err error) {
```

Comment:
Actually, thought about it some more: I'm in favor of everything based off of proto3 to just use the simple Begin, Commit, Rollback etc.

Everything that is not based off of p3 is slowly getting deprecated and therefore it's not a huge deal if they're not cleanly named.

I'm going to standardize on the non-2 version unless there are strong objections :)